### PR TITLE
Username in clone HTTP(S) URL on init repo page

### PR DESCRIPTION
**Pull request by sapk**
Implement #2627

captureecran2627

### CODE REVIEW

**Comment by sapk** [comment]
Maybe a another improvement could be to use net/url to represent HTTPS ? In order to be able to edit the username more easely.

File: modules/middleware/repo.go [hunk 1]
```diff
@@ -186,6 +186,8 @@ func RepoAssignment(args ...bool) macaron.Handler {
 			if displayBare {
 				if !ctx.Repo.IsAdmin() {
 					ctx.Flash.Info(ctx.Tr("repo.repo_is_empty"), true)
+				} else { //Has write access and repo is empty
```
Comment (Unknwon) [inline modules/middleware/repo.go]: You can put comment in the next line and add a space after `//`.

File: models/repo.go [hunk 2]
```diff
@@ -452,13 +452,25 @@ func (repo *Repository) cloneLink(isWiki bool) *CloneLink {
...
+		cl.HTTPS = fmt.Sprintf("%s@%s:%s/%s/%s/%s.git", signedUserName, setting.Domain, setting.HttpPort, setting.AppSubUrl, repo.Owner.Name, repoName)
+	} else {
+		cl.HTTPS = fmt.Sprintf("%s:%s/%s/%s/%s.git", setting.Domain, setting.HttpPort, setting.AppSubUrl, repo.Owner.Name, repoName)
+	}
+	cl.HTTPS = fmt.Sprintf("%s://%s", setting.Protocol, path.Clean(cl.HTTPS))
+
 	return cl
 }
 
+// CloneLink returns clone URLs of repository with the signed username in it.
```
Comment (Unknwon) [inline models/repo.go]: Wrong start of comment.

File: models/repo.go [hunk 3]
```diff
@@ -439,7 +439,7 @@ type CloneLink struct {
 	Git   string
 }
 
-func (repo *Repository) cloneLink(isWiki bool) *CloneLink {
+func (repo *Repository) cloneLink(isWiki bool, signedUserName string) *CloneLink {
```
Comment (Unknwon) [inline models/repo.go]: Call `userName` is good enough, `signedUserName` is narrowing the actual usage of this method.

File: models/repo.go [hunk 4]
```diff
@@ -452,13 +452,25 @@ func (repo *Repository) cloneLink(isWiki bool) *CloneLink {
 	} else {
 		cl.SSH = fmt.Sprintf("%s@%s:%s/%s.git", setting.RunUser, setting.SSHDomain, repo.Owner.Name, repoName)
 	}
-	cl.HTTPS = fmt.Sprintf("%s%s/%s.git", setting.AppUrl, repo.Owner.Name, repoName)
+
+	if len(signedUserName) > 0 {
```
Comment (Unknwon) [inline models/repo.go]: You can compose URL path without `userName` and then add it in the front.

**Comment by fnkr** [comment]
Why is this necessary? It makes clone links noninterchangeable, and adds functionality to URLs that has nothing to do with the sense of an URL. If you want to store your access credentials use  [git credential store](https://git-scm.com/docs/git-credential-sto re) or SSH with public keys as these are are designed to store your authentication data.

**Comment by zacheryph** [comment]
I would agree with @fnkr.  The username has no purpose being in the URL.  You don't put it there for ssh URLs.  The only point to this is to save a few keystrokes when not using `git-credential-store`.

**Comment by Unknwon** [comment]
@fnkr thanks for the comments. This PR only changes the init page.

**Comment by fnkr** [comment]
@Unknwon This PR changes the repo page, which is where you get the git url
from.

**Comment by Unknwon** [comment]
> This PR changes the repo page

How?

**Comment by Unknwon** [comment]
This is not a good idea in a second thought, closing...

**Pull request closed without merging**